### PR TITLE
8331619: TabbedPane's contentOpaque, tabsOpaque and setOpaque doesn't work properly in Aqua LAF

### PR DESCRIPTION
**Pull request by kumarabhi006**
JTabbedPane's contentOpaque and tabsOpaque properties are not honored in Aqua L&F. JTabbedPane's content area and tab background color are not as expected when tabbedpane opacity is set to true or false. Fix is to handle the opacity behavior correctly and inline with other LAF as well. 

Existing test `TestBackgroundScrollPolicy.java` failed with the proposed fix and it is updated to run only for linux and windows platform because the content area for tabbedpane is rendered to the width and height of tabbedpane starting from (0, 0) position (https://github.com/openjdk/jdk/blob/cf7c97732320d70de5f5725c920d5c3861a2c9c8/src/java.desktop/macosx/classes/com/apple/laf/AquaTabbedPaneUI.java#L684C16-L684C16) and that leaves no place for tab area behind tabs.

CI testing is green after this test update and link posted in JBS.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331619](https://bugs.openjdk.org/browse/JDK-8331619): TabbedPane's contentOpaque, tabsOpaque and setOpaque doesn't work properly in Aqua LAF (**Bug** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19170/head:pull/19170` \
`$ git checkout pull/19170`

Update a local copy of the PR: \
`$ git checkout pull/19170` \
`$ git pull https://git.openjdk.org/jdk.git pull/19170/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19170`

View PR using the GUI difftool: \
`$ git pr show -t 19170`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19170.diff">https://git.openjdk.org/jdk/pull/19170.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19170#issuecomment-2104036806)

### CODE REVIEW

**Comment by mrserb** [comment]
The Aqua L&F mimics the behavior of the native UI on macOS. Is it possible to change all these properties in the native apps?

**Comment by kumarabhi006** [comment]
> Is it possible to change all these properties in the native apps?

I am not sure whether we can change the tab or content area color in native apps. I will check and update.

File: src/java.desktop/macosx/classes/com/apple/laf/AquaLookAndFeel.java [hunk 1]
```diff
@@ -860,7 +862,7 @@ public Object createValue(UIDefaults defaultsTable) {
             //"TabbedPane.shadow", table.get("controlShadow"),
             //"TabbedPane.darkShadow", table.get("controlDkShadow"),
             //"TabbedPane.focus", table.get("controlText"),
-            "TabbedPane.opaque", useOpaqueComponents,
+            "TabbedPane.opaque", false,
```
Comment (DamonGuy) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaLookAndFeel.java:865]: Looks like `useOpaqueComponents` is normally set to `Boolean.TRUE`. But now we set it to `false`. Why not `Boolean.FALSE`?

File: src/java.desktop/macosx/classes/com/apple/laf/AquaTabbedPaneUI.java [hunk 2]
```diff
@@ -679,8 +691,16 @@ protected void paintContentBorder(final Graphics g, final int tabPlacement, fina
                 break;
         }
 
-        if (tabPane.isOpaque()) {
-            g.setColor(tabPane.getBackground());
+        if ((tabPane.isOpaque() || contentOpaque) && tabPane.getTabCount() > 0) {
```
Comment (alisenchung) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaTabbedPaneUI.java:694]: what is the difference between tabPane.isOpaque() and contentOpaque? Should isOpaque be set when contentOpaque is set?

File: test/jdk/javax/swing/JTabbedPane/TestBackgroundScrollPolicy.java [hunk 3]
```diff
@@ -48,17 +48,19 @@ public static void main(String[] args) throws Exception {
         ROBOT.setAutoDelay(100);
         for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
             System.out.println("Testing L&F: " + laf.getClassName());
-            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
-            try {
-                SwingUtilities.invokeAndWait(() -> createGUI());
-                ROBOT.waitForIdle();
+            if (!laf.getClassName().contains("Aqua")) {
```
[the remark this answers is not in the document]
Comment (alisenchung) [reply test/jdk/javax/swing/JTabbedPane/TestBackgroundScrollPolicy.java:51]: We could probably just exclude macos in the jtreg tags instead. But that would also mean that we can't run the test on macos on a different L&F. Should this test be run on, for example, metal on macos?

**Comment by kumarabhi006** [comment]
@DamonGuy @alisenchung Thanks for reviewing the PR but I will update the suggested fix or responds to them once I verify the behavior for native apps as suggested by @mrserb here https://github.com/openjdk/jdk/pull/19170#issuecomment-2105005574.

**Comment by kumarabhi006** [comment]
> The Aqua L&F mimics the behavior of the native UI on macOS. Is it possible to change all these properties in the native apps?

I couldn't find any ways to change these properties in the native apps. Do you have any idea whether it is possible ?

**Comment by kumarabhi006** [comment]
@mrserb I am going to close this bug as "Not an issue" as I couldn't find any ways to check this behavior.

**Comment by mrserb** [comment]
> @mrserb I am going to close this bug as "Not an issue" as I couldn't find any ways to check this behavior.

sounds good.

**Pull request closed without merging**